File: test-suite/src/test/java/io/micronaut/jsonschema/test/ServingSchemasTest.java
package io.micronaut.jsonschema.test;

import io.micronaut.http.HttpRequest;
import io.micronaut.http.client.HttpClient;
import io.micronaut.http.client.annotation.Client;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

@MicronautTest
class ServingSchemasTest  {
    @Test
    void testGetSchemas(@Client("/") HttpClient client) {
        String result = client.toBlocking().retrieve(HttpRequest.GET("/schemas/llama.schema.json"));
        assertNotNull(result);
        assertTrue(result.contains("A llama."));

        result = client.toBlocking().retrieve(HttpRequest.GET("/schemas/red-winged-blackbird.schema.json"));
        assertNotNull(result);
        assertTrue(result.contains("A species of blackbird with red wings"));
    }
}

File: test-suite-python/src/test/python/micronaut/jsonschema/test/ServingSchemasTest.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.http import HttpRequest
from micronaut.http.client import HttpClient
from micronaut.http.client.annotation import Client
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test


@MicronautTest
class ServingSchemasTest:
    client: Annotated[HttpClient, Inject, Client("/")]

    @Test
    def test_get_schemas(self) -> None:
        result = self.client.toBlocking().retrieve(HttpRequest.GET("/schemas/llama.schema.json"))
        assert result is not None
        assert "A llama." in result

        result = self.client.toBlocking().retrieve(HttpRequest.GET("/schemas/red-winged-blackbird.schema.json"))
        assert result is not None
        assert "A species of blackbird with red wings" in result
